src/filetree.rs

```diff
@@ -303,7 +303,7 @@ pub(crate) fn apply_diff(
         let path = Path::new(pathstr);
         if let Some(parent) = path.parent() {
             // TODO: care about directory modes?  We don't for FAT.
-            std::fs::create_dir_all(parent)?;
+            destdir.ensure_dir_all(parent, 0o755)?;
```

Comment:
Nit: I think you also took care of the `TODO` above with this. If so, it can be dropped.
Also, there is a very confusing `#[allow(dead_code)] // Used for testing` at the top of this function, which judging from this patch I think it isn't really true.

Reply:
Mmm...I think the `TODO` isn't quite fixed, actually it looks like:
```
rpm -qlv efi-filesystem
drwx------    2 root     root                        0 Jan 28  2020 /boot/efi
drwxr-xr-x    2 root     root                        0 Jan 28  2020 /boot/efi/EFI
drwx------    2 root     root                        0 Jan 28  2020 /boot/efi/EFI/BOOT
drwx------    2 root     root                        0 Jan 28  2020 /boot/efi/EFI/fedora
```

So perhaps we should be 0700 in all of these directories.  (But eh, if we do that why not just make the toplevel `/boot/efi` 0700?)


As far as `dead_code` - yeah definitely but let's fix that as a followup?

Reply:
Fine for doing both later, to be honest I thought the new `0o755` was exactly a reaction to the TODO.